service/share/share_test.go

```diff
@@ -56,3 +56,36 @@ func TestService_GetSharesByNamespace(t *testing.T) {
 		})
 	}
 }
+
+func TestService_GetSharesByNamespaceNotFoundInRange(t *testing.T) {
+	serv, root := RandServiceWithSquare(t, 1)
+	root.RowsRoots = nil
+
+	shares, err := serv.GetSharesByNamespace(context.Background(), root, []byte(""))
+	assert.Len(t, shares, 0)
+	require.Error(t, err, "namespaceID not found in range")
+}
+
+func BenchmarkService_GetSharesByNamespace(b *testing.B) {
```

Comment:
Is there also a test to check for the correctness of the shares that are returned via this process? If not, can you please add it?

Reply:
WDYM by "correctness"?

Reply:
`TestService_GetSharesByNamespace` should be modified so that if `expectedShares > 1`, then check both shares[0] and shares[1]. 

We should also expand testing around this method to cover more cases (like overflow of NID into multiple rows, not just 2)

Reply:
Should this be part of this PR?

Reply:
At least the first point, yes bc it's testing for accuracy of *all* shares returned by `GetShares`

Reply:
@vgonkivs

Reply:
@renaynay, I hope I understand you correctly.
Added loop to `TestService_GetSharesByNamespace` to check all returned shares.

Reply:
Perfect.